flytekit/image_spec/default_builder.py

```
@@ -232,6 +233,11 @@ def create_docker_context(image_spec: ImageSpec, tmp_dir: Path):
     else:
         run_commands = ""
 
+    if image_spec.docker_commands:
```

Could we also add some tests for it?

---

We should raise an exception in `envd` builder when users specify `docker_command` since `envd` doesn't support it.

---

Unit tests of creating docker content and building image have been added, and also the exception for `envd` builder. Do they look good?